.github/workflows/ci-test-xtgeo-cibuildwheel.yml

```
@@ -8,23 +8,31 @@ on:
 
 jobs:
   build_and_upload_with_cibw:
-    name: CIBW python ${{ matrix.cibw_python }} on ${{ matrix.os.runs_on }}
-    runs-on: ${{ matrix.os.runs_on }}
+    name: CIBW python ${{ matrix.cibw_python }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         cibw_python: [cp38, cp39, cp310, cp311, cp312]
-        os:
-          - runs_on: ubuntu-latest
-            cibw_image: manylinux_x86_64
-          - runs_on: windows-latest
-            cibw_image: win_amd64
-          - runs_on: macos-latest
-            cibw_image: macosx_x86_64
-
-    env:
-      CIBW_BUILD: ${{ matrix.cibw_python }}-${{ matrix.os.cibw_image }}
+        os: [ubuntu-latest, windows-latest, macos-latest, macos-latest-xlarge]
+        exclude:
+          - os: macos-latest-xlarge
+            cibw_python: cp39
```

I think we could avoid the inferrence step by doing something like this instead.

```suggestion
        os_arch: 
          - [ubuntu-latest, manylinux_x86_64]
          - [windows-latest, win_amd64]
          - [macos-latest, macosx_x86_64]
          - [macos-latest-xlarge, macosx_arm64]
        exclude:
          - os_arch: [macos-latest-xlarge, macosx_arm64]
             cibw_python: cp39
```

Then

```yaml
name: CIBW python ${{ matrix.cibw_python }} on ${{ matrix.os_arch[0] }}
runs-on: ${{ matrix.os_arch[0] }}
```

```yaml
env:
  CIBW_BUILD: ${{ matrix.cibw_python }}-${{ matrix.os_arch[1] }}
```

I think this might be a bit more maintainable (c.f. https://github.com/orgs/community/discussions/7835#discussioncomment-1769026)